crypto/src/base58/encodable.rs

```diff
@@ -20,17 +18,105 @@ pub trait Base58Encodable {
     {
         let mut buf = Vec::<u8>::new();
         to_writer(&mut buf, self).unwrap();
-        encode(buf).with_check_version(Self::VERSION_BYTE)
+        bs58::encode(buf).with_check_version(Self::VERSION_BYTE)
     }
 
-    fn from_base58<'a, I>(i: I) -> Result<Self, Error>
+    fn to_base58_string(&self) -> String
+    where
+        Self: Sized + Serialize,
+    {
+        self.to_base58().into_string()
+    }
+
+    fn from_base58<'a, I>(i: I) -> Result<Self, bin_prot::error::Error>
     where
         I: AsRef<[u8]>,
         Self: Sized + Deserialize<'a>,
     {
-        let bytes: Vec<u8> = decode(i).with_check(Some(Self::VERSION_BYTE)).into_vec()?;
+        let bytes: Vec<u8> = bs58::decode(i)
+            .with_check(Some(Self::VERSION_BYTE))
+            .into_vec()
+            .map_err(|e| bin_prot::error::Error::Custom {
```

Comment:
I'm not a fan of this use of bin_prot::Error.

Define a new error type for this trait if needed using `thiserror` it much more straightforward and readable

Reply:
Fixed